scripts/globals/abilities/pets/mending_halation.lua

```diff
@@ -0,0 +1,50 @@
+-----------------------------------
+-- Mending Halation
+-- Causes your luopan to vanish and restores HP of party members within area of effect.
+-- Obtained: Geomancer Level 75
+-- Note: Heals for a fixed amount based on Luopan Level:
+-- This is a light-based cure, so it can suffer Darkness day/weather penalties.
+-----------------------------------
+require("scripts/globals/mobskills")
+require("scripts/settings/main")
+require("scripts/globals/status")
+require("scripts/globals/msg")
+-----------------------------------
+local ability_object = {}
+
+ability_object.onAbilityCheck = function(player, target, ability)
+    return 0, 0
+end
+
+ability_object.onPetAbility = function(target, pet, skill)
+    local master    = pet:getMaster()
+    local hpAmount  = math.floor(7 * pet:getMainLvl())
+    local hpRestore = hpAmount
+
+    if master and master:getMerit(xi.merit.MENDING_HALATION) > 0 then
+        hpRestore = hpRestore + (hpRestore * 0.05 * master:getMerit(xi.merit.MENDING_HALATION))
+        if master:getMod(xi.mod.MENDING_HALATION) > 0 then
+            hpRestore = hpRestore + (hpRestore * 0.04 * master:getMerit(xi.merit.MENDING_HALATION))
+        end
+    end
+
+    if (target:getHP()+hpRestore > target:getMaxHP()) then
+        hpRestore = target:getMaxHP() - target:getHP() --cap it
```

Comment:
we actually have clamping functions you can just feed the input to, see utils global

Reply:
and then you won't need the check either, coz it just maxes out.